src/components/UserContainer.tsx

```diff
@@ -1,48 +1,68 @@
-import React from 'react';
+import React, { Component } from 'react';
 import './UserContainer.css';
 import * as Types from '../constants/Types';
 
 type UserContainerProps = {
     displayName: Types.UserDisplay;
-    onClick: (displayName: Types.UserDisplay) => void;
+    onClick?: () => void;
+    onLeaveRoom?: () => void;
     requestSent?: boolean;
     accepted?: boolean;
     currentRoom?: boolean;
 };
 
-const UserContainer: React.FC<UserContainerProps> = ({
-    displayName,
-    onClick,
-    requestSent,
-    accepted,
-    currentRoom,
-}: UserContainerProps) => {
-    const currentRoomNoAction = !accepted && !requestSent && currentRoom;
-    let status = <p></p>;
-    if (accepted) {
-        status = <p className="user-display-status">Accepted</p>;
-    } else {
-        status = requestSent ? <p className="user-display-status">Pending</p> : <p></p>;
-    }
+type UserContainerState = {
+    hovered: boolean;
+}
 
-    return (
-        <div
-            className={`user-container ${currentRoom && 'user-container-current'} ${currentRoomNoAction &&
-                'user-container-initial'}`}
-            onMouseDown={(): void => {
-                onClick(displayName);
-            }}
-        >
-            <div
-                className={`user-display-icon ${currentRoomNoAction && 'user-display-light'}`}
-                style={{ backgroundColor: displayName.color }}
-            ></div>
-            <p className={`user-display-text ${currentRoomNoAction && 'user-display-light'}`}>
-                {displayName.displayName}
-            </p>
-            {status}
-        </div>
-    );
-};
+export default class UserContainer extends Component<UserContainerProps, UserContainerState> {
+    state: UserContainerState = {
+        hovered: false
+    };
 
-export default UserContainer;
+    render(): React.ReactNode {
+        const currentRoomNoAction = !this.props.accepted && !this.props.requestSent && this.props.currentRoom;
+        let status = <p></p>;
+        if (this.props.accepted) {
+            status = <p className="user-display-status">Accepted</p>;
+        } else {
+            status = this.props.requestSent ? <p className="user-display-status">Pending</p> : <p></p>;
+        }
+
+        return (
+            <div
+                className={`user-container ${this.props.currentRoom && 'user-container-current'} ${currentRoomNoAction &&
+                    'user-container-initial'}`}
+                onMouseDown={(): void => {
+                    this.props.onClick && this.props.onClick();
```

Comment:
Why did you make onClick optional?

Reply:
The logged in user container on the top of the left nav bar does not have an `onClick`. Now that I think about it tho, this is not the best way to go about it since that's the only case. But I also don't want to create an entire new component just for that. What do you think a good solution is?

Reply:
Same as the below comment. What about disabling the button and still having the same callback function being passed as a prop?

Reply:
Suree, but don't you think it's weird that you'd be passing a function that is not getting called?